### ISSUE TITLE
Image relation has an unexpected 'related' object that contains duplicate of the root document.

### ISSUE THREAD

**Issue by FieldMarshallVague** [strapi/strapi#13945]
I'm getting a weird response with image population:

When I request an article collection with http://localhost:1337/api/Articles?populate[image][populate]=*

I get back a normal-looking response, but the Image node has a nested 'related' node that includes all my data for the parent Article object!

So, it kind of repeats the Article data in an Image-node property called 'related'.  It means I can't stringify it, because it's a circular reference.  I've checked my Strapi content type and it's just a normal 'Image /Media type' and the API code doesn't create an Image type explicitly, because it's built-in.  Something odd is going on here, but I can't figure out what...

I did upgrade from V3 to V4, is it possibly something to do with that?  I followed guides and used CodeMods back in June to migrate.  I went from v3 to 4.1 to 4.2 to 4.3.beta-1, 4.3.beta-2 and now 4.3.

- Node.js version: 16.14.2
- NPM version: 8.7.0
- Strapi version: 4.3.0
- Database: Postgres
- Operating system: win 10 pro 64

### Describe the bug

This image object (below) appears in an article with ID 2016.  The same ID as the article that appears in the 'related' field of the image object.
I would not expect to find related articles in an image object and I don't know how it got there.  I can see how this might happen, but is this a config option or a bug?  I can't find anything at all the sheds light on it.

```ts
                "image": {
                    "data": {
                        "id": 1381,
                        "attributes": {
                            "name": "/media/70387/style-625px.jpg",
                            "alternativeText": null,
                            "caption": null,
                            "width": 625,
                            "height": 327,
                            "formats": {
                                "small": {
...
                                },
                                "thumbnail": {
...
                                }
                            },
                            "hash": "style_625px_ba1251f301",
                            "ext": ".jpg",
                            "mime": "image/jpeg",
                            "size": 23.6,
                            "url": "https://127.0.0.1:10000/devstoreaccount1/test-strapi-uploads/assets/bhot.jpg",
                            "previewUrl": null,
                            "provider": "azure-storage",
                            "provider_metadata": null,
                            "createdAt": "2022-01-11T11:25:55.250Z",
                            "updatedAt": "2022-01-11T11:25:55.250Z",
                            "related": {
                                "data": [
                                    {
                                        "id": 2016,
                                        "attributes": {
                                            "__type": "api::article.article",
                                            "title": "A Brief History of Time",
                                            "description": "A Brief History of Time ",
                                            "content": "<p>[my article]</p>
                                            "slug": "a-brief-history-of-time",
                                            "updated": null,
                                            "authoredDate": null,
                                            "createdAt": null,
                                            "updatedAt": null,
                                            "publishedAt": "2022-01-11T11:25:34.254Z"
                                        }
                                    }
                                ]
                            }
                        }
                    }
                },

```

**Comment by FieldMarshallVague** [comment]
It might not seem like a big deal, but it's doubling the payload size and stop me stringifying the response (circular reference error).  Also, I don't need it and don't want it in the response.

For now, I'm working around it by being explicit about the population/fields with this config:

```ts
{
            populate: {
              category: {
                fields: ['id'],
              },
              featureDetails: {
                populate: '*',
              },
              image: {
                fields: [
                  'name',
                  'caption',
                  'alternativeText',
                  'width',
                  'height',
                  'url',
                  'formats',
                ],
              },
              relatedArticles: {
                fields: ['id'],
              },
              seo: {
                populate: '*',
              },
            },
            pagination: {
              page: 1,
              pageSize: 999,
            },
          }
```

**Comment by WalkingPizza** [comment]
Quick update about this – that is not a related article, but a list of entities in which the image you are retrieving is being used.

**Comment by FieldMarshallVague** [comment]
Thanks, I'm aware.  It's just unnecessary and confusing.  Having to be explicit about removing it when the image itself is explicitly requested for *inclusion* is counter-intuitive.

**Comment by petersg83** [comment]
Hello :) Why do you populate the fields inside the image?
Does `http://localhost:1337/api/Articles?populate[0]=image` do what you want?

**Comment by FieldMarshallVague** [comment]
@petersg83 Hey :)  It was the only way I could figure out to get all the other child objects as well as ONLY the image fields I wanted, excluding the nested 'related' articles.

The new API response design seems to indicate that we have to include, not exclude data, except in this case.  It's weird.

**Comment by petersg83** [comment]
Ok thank you. And does `http://localhost:1337/api/Articles?populate[0]=image` does what you need? instead of `http://localhost:1337/api/Articles?populate[image][populate]=*`

**Comment by petersg83** [comment]
What about `http://localhost:1337/api/Articles?populate=*`? As described [here](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest/populating-fields.html#relation-media-fields). It will populate the whole first level.

**Comment by FieldMarshallVague** [comment]
No, sorry, that ignores the child ID fields filtering and returns the whole object(s).  So I get all the category data as well as the related articles (as a direct descendant of the article itself, not the image-related articles that this issue is about).

For reference, I omit the full child-object data because I've already pulled that data down into stores, to reduce round trips and 'compose' the data during static site build (Nuxt), I simply map using the ID fields when I need it.  I believe this was the intended use-case for the new JSON:API structure?  If this isn't the intention, then I have no idea what's going on! :D

**Comment by petersg83** [comment]
Ok so I guess everything is working as expected no?
What you do here: https://github.com/strapi/strapi/issues/13945#issuecomment-1202732460 is exactly what you need no?
I'm sorry I don't understand well where the bug would be :p

**Comment by FieldMarshallVague** [comment]
That's OK, it's confusing to me, too.

The 'issue' is that I have to explicitly include child objects EXCEPT the image.related one.  That's it.  The 'workaround' is my config, the 'fix' is making it not include the .related items unless populate is used on the parent.  At least, as far as my interpretation of the Strapi docs goes.  I mean, it's copying the whole of the related articles into the image object... that can't be right.  It's the opposite of the painful structural changes Strapi has made to the response object.

So, my issue was the confusion around the unexpected inclusion of related items on Image.  It doesn't seem right.  Whether this is working correctly or not is not for me to answer, but it doesn't seem right.

**Comment by petersg83** [comment]
Would you be available on discord at some point today?
You can email me at pierre.noel@strapi.io :)

**Comment by petersg83** [comment]
Hello :)
For people arriving on this issue. We discussed and here is what we found:
When using the `populate` param with an object (and not an array of strings), the way to populate a field without any additional parameters (`fields`, `populate`) is to do like that:
```js
populate: {
    category: {
      fields: ['id'],
    },
    image: '*', // <-- here
  },
  pagination: {
    page: 1,
    pageSize: 999,
  },
}
```

On my side I find it strange as `'*'` means to me "populate everything". I would have seen `true` instead. So I'll ask around and come back.
